File: 2025/ts/src/day-9/part_1.ts
import * as fs from 'fs';
import * as readline from 'readline';
import { PATH } from './common';

export async function day9Part1(): Promise<void> {
  try {
    const fileStream = fs.createReadStream(PATH);
    const rl = readline.createInterface({
      input: fileStream,
      crlfDelay: Infinity,
    });


    for await (const line of rl) {
      const trimmedLine = line.trim();
      console.log(trimmedLine);
    }

    const solution = 0;
    console.log(`Solution: ${solution}`);
  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  }
}

File: 2025/python/day_9/part_1.py
import sys
from .common import FILE_PATH

def day9_part1() -> None:
    try:

        with open(FILE_PATH, "r") as f:
            for line in f:
                line = line.strip()

        solution = 0

        print(f"Solution: {solution}")

    except FileNotFoundError:
        print("File not found")
        sys.exit(1)
    except ValueError as e:
        print(f"Parsing error: {e}")
        sys.exit(1)
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)